pkg/await/apps_statefulset_test.go

```diff
@@ -0,0 +1,1252 @@
+        "selector": {
+            "matchLabels": {
+                "app": "foo"
+            }
+        },
+		"serviceName": "%s",
+        "template": {
+            "metadata": {
+                "labels": {
+                    "app": "foo"
+                }
+            },
+            "spec": {
+                "containers": [
+                    {
+                        "name": "nginx",
+                        "image": "nginx",
+						"volumeMounts": [
+							{
+								"mountPath": "/usr/share/nginx/html",
+								"name": "www"
+							}
+						]
+                    }
+                ],
+				"terminationGracePeriodSeconds": 10
+            }
+        },
+		"volumeClaimTemplates": [
+			{
+				"metadata": {
+					"name": "www"
+				},
+				"spec": {
+					"accessModes": [
+						"ReadWriteOnce"
+					],
+					"resources": {
+						"requests": {
+							"storage": "1Gi"
+						}
+					}
+				}
+			}
+		]
+    },
+	"status": {
+		"replicas": 2,
+		"collisionCount": 0,
+		"currentReplicas": 2,
+		"currentRevision": "foo-7b5cf87b78",
+		"observedGeneration": 1,
+		"updateRevision": "foo-7b5cf87b78",
+		"readyReplicas": 1
+    }
+}`, namespace, name, targetService))
+	if err != nil {
+		panic(err)
+	}
+	return obj
+}
+
+func statefulsetReady(namespace, name, targetService string) *unstructured.Unstructured {
+	obj, err := decodeUnstructured(fmt.Sprintf(`{
+    "kind": "StatefulSet",
+    "apiVersion": "apps/v1beta1",
+    "metadata": {
+        "namespace": "%s",
+        "name": "%s",
+		"generation": 1,
+        "labels": {
+            "app": "foo"
+        }
+    },
+    "spec": {
+        "replicas": 2,
+        "selector": {
+            "matchLabels": {
+                "app": "foo"
+            }
+        },
+		"serviceName": "%s",
+        "template": {
+            "metadata": {
+                "labels": {
+                    "app": "foo"
+                }
+            },
+            "spec": {
+                "containers": [
+                    {
+                        "name": "nginx",
+                        "image": "nginx",
+						"volumeMounts": [
+							{
+								"mountPath": "/usr/share/nginx/html",
+								"name": "www"
+							}
+						]
+                    }
+                ],
+				"terminationGracePeriodSeconds": 10
+            }
+        },
+		"volumeClaimTemplates": [
+			{
+				"metadata": {
+					"name": "www"
+				},
+				"spec": {
+					"accessModes": [
+						"ReadWriteOnce"
+					],
+					"resources": {
+						"requests": {
+							"storage": "1Gi"
+						}
+					}
+				}
+			}
+		]
+    },
+	"status": {
+		"replicas": 2,
+		"collisionCount": 0,
+		"currentReplicas": 2,
+		"currentRevision": "foo-7b5cf87b78",
+		"observedGeneration": 1,
+		"updateRevision": "foo-7b5cf87b78",
+		"readyReplicas": 2
+    }
+}`, namespace, name, targetService))
+	if err != nil {
+		panic(err)
+	}
+	return obj
+}
+
+func statefulsetUpdate(namespace, name, targetService string) *unstructured.Unstructured {
```

Comment:
Is this supposed to be the user-provided inputs that update `StatefulSet`, or the `StatefulSet` after the new user inputs are provided? I believe it's the first, in which case is it intentional that `.status` is missing?